packages/ember-data/lib/serializers/rest_serializer.js

```diff
@@ -487,20 +490,22 @@ var RESTSerializer = JSONSerializer.extend({
     var payload = this.normalizePayload(rawPayload);
 
     for (var prop in payload) {
-      var typeName = this.typeForRoot(prop);
-      if (!store.modelFactoryFor(typeName, prop)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, typeName), false);
-        continue;
-      }
-      var type = store.modelFor(typeName);
-      var typeSerializer = store.serializerFor(type);
+      if (prop !== "meta") {
```

Comment:
I strongly disagree with that... the following code failing would just be... unacceptable.

````
this.store.pushPayload("message", data);
if(data.meta !== undefined) {
  this.flash.notice("Only " + data.meta.total + " messages left!");
}
````

That's the kind of bug that makes developers pull their hair out. I don't think saying that this side-effect exists in other parts of the system is a good defense for why it should get worse.

The "if-statement" method in my code works well, because the internal serializer code can mutate it's own data object all it wants, and, while it's slightly slower than deletion, the `pushPayload` method of the serializer is there for public API usage, and mutations would be highly unexpected. This seems like a good trade off to me. 

Also, I haven't observed any mutations in the `pushPayload` serializer in my usage of it, yet. I think those parts of the code might be closer to the non-manual methods.